main-actions/src/main/scala/sbt/Package.scala

```diff
@@ -67,24 +89,30 @@ object Package {
     }
     setVersion(main)
 
-    type Inputs = Map[File, String] :+: FilesInfo[ModifiedFileInfo] :+: Manifest :+: HNil
+    type Inputs = Seq[(File, String)] :+: FilesInfo[ModifiedFileInfo] :+: Manifest :+: HNil
     val cachedMakeJar = inputChanged(cacheStoreFactory make "inputs") {
       (inChanged, inputs: Inputs) =>
         import exists.format
         val sources :+: _ :+: manifest :+: HNil = inputs
         outputChanged(cacheStoreFactory make "output") { (outChanged, jar: PlainFileInfo) =>
           if (inChanged || outChanged) {
-            makeJar(sources.toSeq, jar.file, manifest, log)
+            makeJar(sources, jar.file, manifest, log)
             jar.file
           } else
             log.debug("Jar uptodate: " + jar.file)
         }
     }
 
-    val map = conf.sources.toMap
```

Comment:
This was the actual fix

Reply:
An interesting effect the use of `Map` had was deduplicating the input mapping.
https://github.com/sbt/sbt/issues/4889